server/src/test/java/org/opensearch/index/mapper/ObjectMapperTests.java

```diff
@@ -178,6 +178,32 @@ public void testFieldsWithFilledArrayShouldThrowException() throws Exception {
         }
     }
 
+    public void testDottedFields() throws Exception {
+        String mapping = Strings.toString(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("properties")
+                .startObject(".")
+                .field("type", "text")
+                .field("dimension", 4)
+                .startObject("method")
+                .field("name", "hnsw")
+                .field("space", "l2")
+                .field("engine", "lucene")
+                .endObject()
```

Comment:
Nitpick - you don't really need these extra fields. A mapping of 

```
{ "properties": { "." { "type": "text" }}}
```

is sufficient to capture/reproduce the issue.